src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -76,7 +75,16 @@ protected bool NeedsGeneratedNullableAttribute
             get
             {
                 _needsGeneratedAttributes_IsFrozen = true;
-                return Compilation.NeedsGeneratedNullableAttribute || _needsGeneratedNullableAttribute_Value;
+                if (Compilation.NeedsGeneratedNullableAttribute)
+                {
+                    return true;
+                }
+                if (SourceModule.UtilizesNullableReferenceTypes)
```

Comment:
>SourceModule.UtilizesNullableReferenceTypes [](start = 20, length = 43)

It feels wrong that we should check this explicitly here, I think ```Compilation.NeedsGeneratedNullableAttribute``` should cover this.